core/src/main/java/com/datastrato/gravitino/catalog/rel/BaseTable.java

```diff
@@ -4,19 +4,23 @@
  */
 package com.datastrato.gravitino.catalog.rel;
 
+import com.datastrato.gravitino.exceptions.NoSuchPartitionException;
+import com.datastrato.gravitino.exceptions.PartitionAlreadyExistsException;
 import com.datastrato.gravitino.meta.AuditInfo;
 import com.datastrato.gravitino.rel.Column;
-import com.datastrato.gravitino.rel.Table;
+import com.datastrato.gravitino.rel.PartitionChange;
+import com.datastrato.gravitino.rel.SupportsPartitions;
 import com.datastrato.gravitino.rel.expressions.distributions.Distribution;
 import com.datastrato.gravitino.rel.expressions.sorts.SortOrder;
 import com.datastrato.gravitino.rel.expressions.transforms.Transform;
+import com.datastrato.gravitino.rel.partitions.Partition;
 import java.util.Map;
 import javax.annotation.Nullable;
 import lombok.ToString;
 
 /** An abstract class representing a base table in a relational database. */
 @ToString
-public abstract class BaseTable implements Table {
+public abstract class BaseTable implements SupportsPartitions {
```

Comment:
Should `BaseTable` support `SupportsPartitions`?

Reply:
There is a default unsupported implementation, so the underlying catalog does not need to implement this interface repeatedly.

Reply:
I think this is rebundant. Because we already have a default implement method `supportPartitions` in the interface `Table`.

Reply:
removed